### ISSUE TITLE
SQL Mapper Extensions GetFormatter returns incorrect adapter when using Transacted Connection

### ISSUE THREAD

**Issue by AteCoder** [StackExchange/Dapper#1478]
In the code below, the 'name' is set to TransactedConnection, which, naturally, does not have an entry in the AdapterDictionary, returning the default SQL Server adapter. Most likely, an exception is needed for a Transacted Connection getting the connection name from the Conn property of the Transacted Connection

`        private static ISqlAdapter GetFormatter(IDbConnection connection)
        {
            var name = GetDatabaseType?.Invoke(connection).ToLower()
                       ?? connection.GetType().Name.ToLower();

            return AdapterDictionary.TryGetValue(name, out var adapter)
                ? adapter
                : DefaultAdapter;
        }
`

**Comment by mgravell** [comment]
The problem there is: `TransactedConnection` is something that only exists in the tests, not the library. There is no well-known "I'm a decorator" pattern, for `DbConnection`, or in general. Are you using `TransactedConnection` somewhere?

**Comment by AteCoder** [comment]
Yes, I am using the Transacted Connection thinking it is a super clever way to implement a transaction on existing code.
________________________________

**Comment by AteCoder** [comment]
Thank u for the quick response and for the great work u guys are putting into Dapper. It is much appreciated!

Seeing how Transacted Connection allowed me to use transactions on existing code, Transacted Connections seem like a feature many can use.

Meanwhile, it has become part of my fork of the Library.

Also in that fork are:
- removing the 's suffix to table names;
- support for table inserts with 'default' values as with a table of an auto increment key and default timestamp.

Another issue I had to deal with is the Library defaulting to assume a field named 'id' is a key field where in fact I used it as a foreign key to another table. Renaming the field solved that issue. Perhaps it might be helpful to used an attribute called 'field' for
designating  a simple field.

Also, I was wondering if the Library could benefit from a set of settings such as 'require attribute' for keys and optional appending of 's to table names. In fact, supporting plurality for table names requires language-specific dictionaries. Ergo my preference for dropping plurality altogether.
